packages/cli/src/constructs/construct.ts

```diff
@@ -4,26 +4,75 @@ import { Diagnostics } from './diagnostics'
 import { Session } from './project'
 import { Ref } from './ref'
 
+/**
+ * Base interface for construct-like objects in the Checkly CLI system.
+ * Used for type checking and identifying construct objects.
+ */
 export interface ConstructLike {
+  /** The type identifier of the construct */
   type: string
+  /** Unique logical identifier within the project scope */
   logicalId: string
 }
 
+/**
+ * Interface for objects that can be bundled into a synthesizable representation.
+ * Provides the contract for converting constructs into their deployable form.
+ */
 export interface Bundle {
+  /** Synthesizes the construct into its deployable representation */
   synthesize (): any | null
 }
 
+/**
+ * Interface for constructs that can be validated.
+ * Provides the contract for validating construct configuration and reporting issues.
+ */
 export interface Validate {
+  /** 
+   * Validates the construct, reporting any issues via the provided diagnostics.
+   * @param diagnostics The diagnostics instance to add any validation issues to
+   */
   validate (diagnostics: Diagnostics): Promise<void>
 }
 
+/**
+ * Abstract base class for all constructs in the Checkly CLI system.
+ * Provides common functionality for validation, bundling, and resource management.
+ * 
+ * @example
+ * ```typescript
+ * class MyConstruct extends Construct {
+ *   constructor(logicalId: string) {
+ *     super('my-construct', logicalId);
+ *   }
+ *   
+ *   synthesize() {
+ *     return { name: 'my-construct' };
+ *   }
+ * }
+ * ```
```

Comment:
I don't think we want to provide an example for this. While technically a customer may be able to create their own constructs, they should instead extend one of the higher level types such as `ApiCheck`. If we openly support creating custom constructs there's no telling what will happen over time.

```suggestion
```